public/components/register-agents.js

```diff
@@ -0,0 +1,290 @@
+            </EuiCodeBlock>
+          </EuiText>
+        ) : (<EuiFlexItem />)}
+        {this.state.selectedOS === 'deb' ? (
+          <EuiText>
+            <p>Install the necessaries utilities</p>
+            <div style={copyButton}>
+              <EuiCopy textToCopy={debText} >
+                {copy => (
+                  <EuiButtonIcon
+                    onClick={copy}
+                    iconType="copy"
+                    aria-label="Copy"
+                  />
+                )}
+              </EuiCopy>
+            </div>
+            <EuiCodeBlock style={codeBlock} language="js">
+              {debText}
+            </EuiCodeBlock>
+            <p>Install the Wazuh repository GPG key</p>
+            <div style={copyButton}>
+              <EuiCopy textToCopy={debText2} >
+                {copy => (
+                  <EuiButtonIcon
+                    onClick={copy}
+                    iconType="copy"
+                    aria-label="Copy"
+                  />
+                )}
+              </EuiCopy>
+            </div>
+            <EuiCodeBlock style={codeBlock} language="js">
+              {debText2}
+            </EuiCodeBlock>
+            <p>Add the repository</p>
+            <div style={copyButton}>
+              <EuiCopy textToCopy={debText3} >
+                {copy => (
+                  <EuiButtonIcon
+                    onClick={copy}
+                    iconType="copy"
+                    aria-label="Copy"
+                  />
+                )}
+              </EuiCopy>
+            </div>
+            <EuiCodeBlock style={codeBlock} language="js">
+              {debText3}
+            </EuiCodeBlock>
+            <p>Deploy the installation</p>
+            <div style={copyButton}>
+              <EuiCopy textToCopy={debText4} >
+                {copy => (
+                  <EuiButtonIcon
+                    onClick={copy}
+                    iconType="copy"
+                    aria-label="Copy"
+                  />
+                )}
+              </EuiCopy>
+            </div>
+            <EuiCodeBlock style={codeBlock} language="js">
+              {debText4}
+            </EuiCodeBlock>
+          </EuiText>
+        ) : (<EuiFlexItem />)}
+        {this.state.selectedOS === 'win' ? (
+          <EuiText>
+            <p>Deploy the installation</p>
+            <div style={copyButton}>
+              <EuiCopy textToCopy={winText} >
+                {copy => (
+                  <EuiButtonIcon
+                    onClick={copy}
+                    iconType="copy"
+                    aria-label="Copy"
+                  />
+                )}
+              </EuiCopy>
+            </div>
+            <EuiCodeBlock style={codeBlock} language="js">
+              {winText}
+            </EuiCodeBlock>
+          </EuiText>
+        ) : (<EuiFlexItem />)}
+        {this.state.selectedOS === 'macos' ? (
+          <EuiText>
+            <p>Deploy the installation</p>
```

Comment:
Simplify all this logic, so much duplicated blocks, they differ just in a string